core/primitives/src/receipt.rs

```diff
@@ -55,34 +55,242 @@ pub struct Receipt {
     pub receipt: ReceiptEnum,
 }
 
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    Debug,
+    PartialEq,
+    Eq,
+    Clone,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct ReceiptV1 {
+    /// An issuer account_id of a particular receipt.
+    /// `predecessor_id` could be either `Transaction` `signer_id` or intermediate contract's `account_id`.
+    pub predecessor_id: AccountId,
+    /// `receiver_id` is a receipt destination.
+    pub receiver_id: AccountId,
+    /// An unique id for the receipt
+    pub receipt_id: CryptoHash,
+    /// A receipt type
+    pub receipt: ReceiptEnum,
+    /// Priority of a receipt
+    pub priority: u64,
```

Comment:
could we also wrap this with a struct such as 
pub struct Priority(u64)
to not use u64 directly but the surrounding type?
also is 0 top priority and the larger the number the less priority? can you add comment on this?

Reply:
yeah that is a good point. How exactly priority works has not been decided.

Reply:
I think I'll address this issue in a following PR as this one is quite large and is difficult to merge into master due to conflicts.